pkg/apis/alicloud/validation/infrastructure.go

```diff
@@ -97,8 +100,48 @@ func ValidateInfrastructureConfig(infra *apisalicloud.InfrastructureConfig, node
 func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *apisalicloud.InfrastructureConfig) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	var (
+		newZonesWithoutNatGatewayConfig = make([]apisalicloud.Zone, 0, len(newConfig.Networks.Zones))
+		oldZonesWithoutNatGatewayConfig = make([]apisalicloud.Zone, 0, len(oldConfig.Networks.Zones))
+
+		networksPath = field.NewPath("networks")
+	)
+
+	for i, zone := range newConfig.Networks.Zones {
+		newZonesWithoutNatGatewayConfig = append(newZonesWithoutNatGatewayConfig, apisalicloud.Zone{
+			Name:    zone.Name,
+			Worker:  zone.Worker,
+			Workers: zone.Workers,
+		})
+
+		allErrs = append(allErrs, ValidateNatGatewayConfig(zone.NatGateway, networksPath.Child("zones").Index(i).Child("natGateway").Child("ipAddress"))...)
+	}
+
+	for i, zone := range oldConfig.Networks.Zones {
+		oldZonesWithoutNatGatewayConfig = append(oldZonesWithoutNatGatewayConfig, apisalicloud.Zone{
+			Name:    zone.Name,
+			Worker:  zone.Worker,
+			Workers: zone.Workers,
+		})
+
+		allErrs = append(allErrs, ValidateNatGatewayConfig(zone.NatGateway, networksPath.Child("zones").Index(i).Child("natGateway").Child("ipAddress"))...)
+	}
+
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newConfig.Networks.VPC, oldConfig.Networks.VPC, field.NewPath("networks").Child("vpc"))...)
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newConfig.Networks.Zones, oldConfig.Networks.Zones, field.NewPath("networks").Child("zones"))...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newZonesWithoutNatGatewayConfig, oldZonesWithoutNatGatewayConfig, field.NewPath("networks").Child("zones"))...)
+
+	return allErrs
+}
+
+func ValidateNatGatewayConfig(natGateway *apisalicloud.NatGatewayConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if natGateway != nil && natGateway.IPAddress != nil {
+		ipAddr := string(*natGateway.IPAddress)
+		if ipAddrParsed := net.ParseIP(ipAddr); ipAddrParsed == nil {
```

Comment:
Hm, out of curiosity - why not let people specify the EIP ID instead of the IP itself?

Reply:
Hm, I think `IP address` is more readable and understandable compared with `EIP ID`, so let people able to configure the ip address. Also it would be more convenient for people to retrieve the ip address. (directly from gardener dashboard, not necessary to login Alicloud console).

Of course, if let people specify `EIP ID`, our code can directly bind this value on `vswitch` in the TF scripts, no need to obtain it from `data` reosource.

Reply:
Yeah, just saying that it's inconsistent now, as people would need to specify a VPC _ID_ instead of a _name_ when they want to re-use an existing VPC. But it's your decision, just got curious. :)

Reply:
ah...good point! I missed that it would be inconsistent with re-using an existing VPC feature. I will modify my code, and make the behavior still consistent. :)

Reply:
Maybe you can allow both :)